oe_contact_forms.module

```diff
@@ -537,3 +539,30 @@ function oe_contact_forms_mail_alter(&$message) {
   // Change subject for corporate mail, copy and reply.
   $message['subject'] = $contact_form->getThirdPartySetting('oe_contact_forms', 'email_subject');
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_create_access().
+ */
+function oe_contact_forms_contact_message_create_access(AccountInterface $account, array $context, $entity_bundle) {
+  if (!$entity_bundle) {
+    return AccessResult::neutral();
+  }
+
+  /** @var \Drupal\contact\Entity\ContactForm $contact_form */
+  $contact_form = ContactForm::load($entity_bundle);
+
+  if (!$contact_form) {
+    return AccessResult::neutral();
+  }
+
+  // Leave default contact forms untouched.
+  if (!$contact_form->getThirdPartySetting('oe_contact_forms', 'is_corporate_form', FALSE)) {
+    return AccessResult::neutral()->addCacheableDependency($contact_form);
+  }
+
+  if ($account->hasPermission('access corporate contact form')) {
+    return AccessResult::allowed()->cachePerPermissions();
```

Comment:
You are missing the cache tags on the corporate form. If access was granted because it was a corporate form and then it stops being a corporate form, you had cached only by permissions.

Reply:
I've added the contact_form cache tags